src/devices/efergy_optical.c

```diff
@@ -0,0 +1,139 @@
+/* Efergy e2 classic (electricity meter)
+ *
+ * This electricity meter periodically reports current power consumption
+ * on frequency ~433.55 MHz. The data that is transmitted consists of 8
+ * bytes:
+ *
+ * Byte 1-4: Start bits (0000), then static data (probably device id)
+ * Byte 5-7: all zeros 
+ * Byte 8: Pulse Count
+ * Byte 9: sample frequency (15 seconds)
+ * Byte 10: seconds
+ * Byte 11: crc16 xmodem
+ * Byte 12: crc16 xmodem
+ * if pulse count <3 then energy =(( pulsecount/impulse-perkwh) * (3600/seconds))
+ * else energy = ((pulsecount/n_imp) * (3600/seconds))
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+#include "rtl_433.h"
+#include "util.h"
+#include "data.h"
+
+static int efergy_optical_callback(bitbuffer_t *bitbuffer) {
+	unsigned num_bits = bitbuffer->bits_per_row[0];
+	uint8_t *bytes = bitbuffer->bb[0];
+	double energy, n_imp;
+	double pulsecount;
+	double seconds;
+	data_t *data;
```

Comment:
Probably not, as I am just blundering way through. So please feel free to withhold the merge until I get the code more presentable.  I guess I was just a bit too eager. Most of the code was taken from the efergy e2 classic including the callback limits.
On 7 Mar 2016 11:21 p.m., Benjamin Larsson <notifications@github.com> wrote:
>
> In src/devices/efergy_optical.c:
>
> > + * it under the terms of the GNU General Public License as published by
>
> > + * the Free Software Foundation; either version 2 of the License, or
>
> > + * (at your option) any later version.
>
> > + */
>
> > +
>
> > +#include "rtl_433.h"
>
> > +#include "util.h"
>
> > +#include "data.h"
>
> > +
>
> > +static int efergy_optical_callback(bitbuffer_t *bitbuffer) {
>
> > +	unsigned num_bits = bitbuffer->bits_per_row[0];
>
> > +	uint8_t *bytes = bitbuffer->bb[0];
>
> > +	double energy, n_imp;
>
> > +	double pulsecount;
>
> > +	double seconds;
>
> > +	data_t *data;
>
> Do you really need double for the variables ?
>
> —
> Reply to this email directly or view it on GitHub.